src/SFML/Graphics/Shader.cpp

```diff
@@ -40,6 +40,18 @@
 
 #ifndef SFML_OPENGL_ES
 
+#if defined(SFML_SYSTEM_MACOS) || defined(SFML_SYSTEM_IOS)
+
+    #define castToGlHandle(x) reinterpret_cast<GLEXT_GLhandle>(static_cast<ptrdiff_t>(x))
+    #define castFromGlHandle(x) static_cast<unsigned int>(reinterpret_cast<ptrdiff_t>(x))
+
+#else
+
+    #define castToGlHandle(x) (x)
+    #define castFromGlHandle(x) (x)
+
+#endif
```

Comment:
Actually, do you even need the case differentiation at all? Can't you just use `reinterpret_cast()` in all cases, even if input and output are identical? Never tried that to be honest, but don't see why it shouldn't work.